File: src/main/java/com/example/interviewlandbackend/dto/response/ContentDto.java
package com.example.interviewlandbackend.dto.response;


// Response classlarinda is vacib olan get methodlaridir . Bos constructor olmasi vacib deyil.
public class ContentDto {


    private int id;

    private String contentName;



    public ContentDto(int id , String contentName) {
        this.id = id;
        this.contentName = contentName;
    }


    public int getId() {
        return id;
    }

    public String getContentName() {
        return contentName;
    }


}

File: src/main/java/com/example/interviewlandbackend/dto/response/ContentDto.kt
package com.example.interviewlandbackend.dto.response

import com.example.interviewlandbackend.model.Content

data class ContentDto @JvmOverloads constructor(

    val id: Int?,
    val contentName: String,
    val description : String,
    val imageDto: ImageDto

){

    companion object{

        @JvmStatic fun convert(content: Content):ContentDto{
            return ContentDto(
                content.id ,
                content.contentName ,
                content.description ,
                ImageDto.convert(content.image)
            )
        }

    }

}
